bin/dev-restore-db

```diff
@@ -29,21 +29,27 @@ TRUNCATE_SQL="${TRUNCATE_SQL//$'\n'/}"
 # location of dump file in docker container
 readonly DUMP_FILE="project/${1}"
 
-# command for invoking psql with the dev database
-readonly TRUNCATE_DB_CMD="/usr/bin/psql -h db -U postgres -c '${TRUNCATE_SQL}' postgres"
+readonly COMPOSE_CMD="docker compose \
+  -f docker-compose.yml \
+  -f docker-compose.dev.yml"
 
-# command for invoking pg_restore with the dev database
+docker::set_network_name_dev
 
-readonly PG_RESTORE_CMD="/usr/bin/pg_restore -h db -U \
-  postgres -d postgres < ${DUMP_FILE}"
+${COMPOSE_CMD} up db \
+  --no-deps \
+  --wait \
+  -d
 
-readonly CMD="${TRUNCATE_DB_CMD} && ${PG_RESTORE_CMD}"
+${COMPOSE_CMD} cp ${1} db:/${DUMP_FILE}
 
-docker::set_network_name_dev
+# command for invoking psql with the dev database
+${COMPOSE_CMD} \
+  exec -it -e "PGPASSWORD=example" db \
+  /usr/bin/psql -h db -U postgres \
+  -c "${TRUNCATE_SQL}" postgres
 
-docker run --rm -it \
-  -v "$(pwd):/project" \
-  -e "PGPASSWORD=example" \
-  --network "${DOCKER_NETWORK_NAME}" \
-  postgres:12.5 \
-  /bin/bash -c "${CMD}"
+# command for invoking pg_restore with the dev database
+${COMPOSE_CMD} \
+  exec -it -e "PGPASSWORD=example" db \
+  /usr/bin/pg_restore -h db -U postgres \
+   -f "${DUMP_FILE}"
```

Comment:
this isn't covered by a test, so please confirm the script still works with manual testing

Reply:
confirmed